### PR TITLE
Rewrite CHANGELOG to follow keepachangelog format

### PR DESCRIPTION
**Pull request by igordejanovic**
Also updated/added all notable changes after v1.8.0.

refs #171

## Code review checklist

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [ ] Changelog(s) has/have been updated, as needed (see `CHANGELOG.md`).

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by igordejanovic** [comment]
@goto40 Please read the new CHANGELOG. Also, a new template for PRs is in place that should help in raising the quality of our code and git log history. The idea is that reviewer should check and tick all checks boxes above while doing the review.

**Comment by goto40** [comment]
I will do it. Good checklist!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,35 +1,68 @@
...
-     https://github.com/textX/textX/issues/117
-
-
-* 2018-10-06 Release 1.8.0
+All _notable_ changes to this project will be documented in this file.
+
+The format is based on _[Keep a Changelog][keepachangelog]_, and this project
+adheres to _[Semantic Versioning][semver]_.
+
+Everything that is documented in the [official docs][textXDocs] is considered
```
Comment (goto40) [inline CHANGELOG.md:8]: In principal this is OK for me. That is how I understand the responsibility of official docs. However, 'everything' is a hard word. Also all referenced unittests? With all details? 

We should at least exclude _underscore properties/functions, except the _tx_* aspects. This gives us the ability to mark implementation details in the future.

Comment (goto40) [reply CHANGELOG.md:8]: Again, in principal it is OK for me! We should try hard to document all changes! So maybe ignore the last comment.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,35 +1,68 @@
...
+
+  - CLI migrated to the [click] library ([#162])
+  - Docs improvements ([#146], [#153], [#151]). Thanks simkimsia@GitHub.
+  - `FQN` constuctor param `follow_loaded_models` removed and introduced
+    callback `scope_rediction_logic` to implement arbitrary logic on FQN
+    resolution ([#133], [#114], [#103]) **(BIC)**
+  - Changed attribute name for the meta-model object (from `metamodel._parser`
+    to `metamodel._parser_blueprint`). ([#93]) **(BIC)**
+  - Started using _[Keep a Changelog][keepachangelog]_ ([#?])
+  - Started using _[Semantic Versioning][semver]_ ([#?])
```
Comment (goto40) [inline CHANGELOG.md]: References with '?'

**Review by goto40 — CHANGES_REQUESTED** [review]
Very good! From my side the checklist is a great add-on.

One thing: Some references have '?'. See inline comments. 

No additional critical comments.

File: CHANGELOG.md (hunk 1 above)
Comment (igordejanovic) [reply CHANGELOG.md:8]: Yup, we should somehow document the what should be considered a stable API for which change we shall increase the major version number. I guess the correct way is to document stuff that are user-facing and whenever we change those element is a backward incompatible way we do major number increase.

We should also make clear the policy we use to maintain old major versions. I guess we should give enough time for users to migrate to new major versions. Maybe to provide bugfixes to the next to last major version. E.g. when we release 2.0.0 we might provide bugfixes for 1.8.x until we hit 3.0.0. That should give plenty of time for the upgrade but will increase maintenance overhead. Not sure that we have a capacity for that.

File: CHANGELOG.md (hunk 2 above)
Comment (igordejanovic) [reply CHANGELOG.md]: Good catch! Yeah, I put a question mark with an idea to replace with a proper PR number as soon as I make one, but forgot to update. Will do.

File: CHANGELOG.md (hunk 1 above)
Comment (igordejanovic) [reply CHANGELOG.md:8]: Even without support for previous version, users should be safe to declare their dependencies against the major version and receive all backward compatible improvements and bug fixes. So they can stay at previous version as long as they like. When the time is to upgrade to the next major version they should investigate the changelog for all breaking changes and see if they are affected. That should provide (I hope) smooth upgrades.

Comment (goto40) [reply CHANGELOG.md:8]: OK for me.

**Review by goto40 — APPROVED** [review]

**Pull request merged**